dotCMS/src/main/java/com/dotcms/rest/BundlePublisherResource.java

```diff
@@ -47,182 +45,140 @@
+			@QueryParam("type") String type,
+			@QueryParam("callback") String callback,
+			@QueryParam("BUNDLE_NAME") String bundleName,
+			@QueryParam("FORCE_PUSH") final boolean forcePush,
+			@Context HttpServletRequest req
+	) {
     	try {
-			return new WebResource.InitBuilder().
-					rejectWhenNoUser(false). // it would be a soft validation so not reject
-					requestAndResponse(
-					new HttpHeaderHandlerHttpServletRequestWrapper(request,
-							CollectionsUtils.map(
-									"Authorization", (name, value) -> // if the authorization is set, uses it, otherwise try with the secret (could be a jwt)
-											UtilMethods.isSet(value) ? value : JsonWebTokenAuthCredentialProcessor.BEARER + authTokenDigest
-							)), response).init();
-		}catch (Exception e) {
-    		return null;
-		}
-
-	}
-
-
-	@WrapInTransaction
-	private Bundle publishBundle(final String fileNameSent,
-								  final String groupId,
-								  final String endpointId,
-								  final String bundleNameSent,
-								  final boolean forcePush,
-								  final HttpServletRequest request,
-								  final String remoteIP,
-								  final PublishingEndPoint sendingEndPointByAddress) throws Exception {
-
-    	final String fileName = UtilMethods.isSet(fileNameSent) ? fileNameSent : generatedBundleFileName();
-		final String bundleName =  UtilMethods.isSet(bundleNameSent) ? bundleNameSent : fileName;
-
-		Bundle bundle = null;
-
-		try (InputStream bundleStream = request.getInputStream()) {
-
-			final String bundlePath         = ConfigUtils.getBundlePath()+ File.separator + MY_TEMP;
-			final String bundleFolder       = fileName.substring(0, fileName.indexOf(".tar.gz"));
-			final String sendingEndPoint = sendingEndPointByAddress != null ? sendingEndPointByAddress.getId() : remoteIP;
-			final PublishAuditStatus status = PublishAuditAPI.getInstance().updateAuditTable(
-					sendingEndPoint, sendingEndPoint, bundleFolder, true);
-
-			if(bundleName.trim().length() > 0) {
-				// save bundle if it doesn't exists
-				bundle = APILocator.getBundleAPI().getBundleById(bundleFolder);
-				if (bundle == null || bundle.getId() == null) {
-
-					bundle = new Bundle();
-					bundle.setId(bundleFolder);
-					bundle.setName(bundleName);
-					bundle.setPublishDate(Calendar.getInstance().getTime());
-					bundle.setOwner(APILocator.getUserAPI().getSystemUser().getUserId());
-					bundle.setForcePush(forcePush);
-					APILocator.getBundleAPI().saveBundle(bundle);
+    		try (InputStream bundleStream = req.getInputStream()) {
+		        //Creating an utility response object
+		        Map<String, String> paramsMap = new HashMap<String, String>();
+		        paramsMap.put( "type", type );
+		        paramsMap.put( "callback", callback );
+		        ResourceResponse responseResource = new ResourceResponse( paramsMap );
+
+				String remoteIP = "";
+				try {
+
+					remoteIP = req.getRemoteHost();
+					if(!UtilMethods.isSet(remoteIP))
+						remoteIP = req.getRemoteAddr();
```

Comment:
Codacy found an issue: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5721924)